public/diagnostic.js

```diff
@@ -23,3 +23,31 @@
 //  -  Order 1 : 2 hats ($5 each)
 //  -  Order 2 : 1 desk lamp ($20 each)
 //  -  Order 3 : 3 hand towels ($8 each)
+import Ember from 'ember';
+
+const Order = Ember.Object.extend({
+  name: 'Default',
+  unitPrice: 0.99,
+  quantity: 1,
+  orderPrice: Ember.computed('unitPrice', 'quantity', function() {
+    return this.get('unitPrice') * this.get('quantity');
+  })
+});
+
+const Cart = Ember.Object.extend({
+  orders: Ember.A([]),
+  addToCart: function (p_name, p_quantity, p_unitPrice) {
+    this.get('orders').pushObject(Order.create({ name: p_name, quantity: p_quantity, p_unitPrice }));
+  },
+  totalPrice: Ember.computed('orders.@each', function() {
+    let orders = this.get('orders');
+    return orders.reduce(function(previousValue, order){
+      return previousValue + (order.get('quantity') * order.get('unitPrice'));
```

Comment:
You already did this work in creating the 'orderPrice' property on orders, so I think it would make more sense to use orderPrice here rather than recalculating it.